pandas/tests/extension/base/setitem.py

```diff
@@ -36,25 +38,27 @@ def test_setitem_sequence_mismatched_length_raises(self, data, as_array):
         with tm.assert_raises_regex(ValueError, xpr.format('slice')):
             ser[slice(3)] = value
 
-    def test_setitem_empty_indxer(self, data):
-        ser = pd.Series(data)
-        original = ser.copy()
-        ser[[]] = []
-        self.assert_series_equal(ser, original)
-
-    def test_setitem_sequence_broadcasts(self, data):
-        arr = pd.Series(data)
+    def test_setitem_empty_indxer(self, data, box):
+        if box:
+            data = pd.Series(data)
+        original = data.copy()
+        data[[]] = []
+        self.assert_series_equal(data, original)
 
-        arr[[0, 1]] = data[2]
-        assert arr[0] == data[2]
-        assert arr[1] == data[2]
+    def test_setitem_sequence_broadcasts(self, data, box):
+        if box:
+            data = pd.Series(data)
+        data[[0, 1]] = data[2]
+        assert data[0] == data[2]
+        assert data[1] == data[2]
 
     @pytest.mark.parametrize('setter', ['loc', 'iloc'])
-    def test_setitem_scalar(self, data, setter):
-        arr = pd.Series(data)
-        setter = getattr(arr, setter)
+    def test_setitem_scalar(self, data, box, setter):
```

Comment:
I think this will not work, since `loc` and `iloc` are only available on Series.

Reply:
changed.
thanks for the quick feedback!